### PR TITLE
fix(security) sanitize host input

### PR DESCRIPTION
**Pull request by loiclau**
## Description

sanitize host input form

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 19.10.x
- [x] 20.04.x
- [x] 20.10.x
- [x] 21.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/configuration/configObject/host/DB-Func.php [hunk 1]
```diff
@@ -1375,169 +1247,23 @@ function updateHost($host_id = null, $from_MC = false, $cfg = null)
...
-        : $rq .= "NULL, ";
-    $rq .= "host_activate = ";
-    isset($ret["host_activate"]["host_activate"]) && $ret["host_activate"]["host_activate"] != null
-        ? $rq .= "'" . $ret["host_activate"]["host_activate"] . "' "
-        : $rq .= "NULL ";
-    $rq .= "WHERE host_id = '" . $host_id . "'";
-    $dbResult = $pearDB->query($rq);
+    $ret["host_name"] = $host->checkIllegalChar($ret["host_name"], $server_id);
+    
+    $bindParams = sanitizeFormContactParameters($ret);
```
Comment (loiclau) [inline www/include/configuration/configObject/host/DB-Func.php:1252]:
```suggestion
    $bindParams = sanitizeFormHostParameters($ret);
```

File: www/include/configuration/configObject/host/DB-Func.php [hunk 2]
```diff
@@ -2858,3 +2465,124 @@ function applytpl($hosts)
...
+            case 'host_name':
+            case 'host_alias':
+            case 'host_address':
+            case 'host_snmp_community':
+            case 'host_snmp_version':
+            case 'host_comment':
+            case 'geo_coords':
+                if (!empty($inputValue)) {
+                    $bindParams[':' . $inputName] = [
+                        \PDO::PARAM_STR => (($inputValue = filter_var($inputValue, FILTER_SANITIZE_STRING)) === false)
```
Comment (sc979) [inline www/include/configuration/configObject/host/DB-Func.php:2513]: SANITIZE_STRING is never false, but could be empty

File: www/include/configuration/configObject/host/DB-Func.php [hunk 3]
```diff
@@ -2858,3 +2465,124 @@ function applytpl($hosts)
...
+                        ? $inputValue[$inputName]
+                        : '2'
+                ];
+                break;
+            case 'host_notifOpts':
+                if (!empty($inputValue)) {
+                    $bindParams[':host_notification_options'] = [
+                        \PDO::PARAM_STR => (($inputValue = filter_var(
+                            implode(",", array_keys($inputValue)),
+                            FILTER_SANITIZE_STRING
```
Comment (sc979) [inline www/include/configuration/configObject/host/DB-Func.php:2540]: SANITIZE_STRING is never false, but could be empty

File: www/include/configuration/configObject/host/DB-Func.php [hunk 4]
```diff
@@ -2858,3 +2465,124 @@ function applytpl($hosts)
...
+                $bindParams[':' . ltrim($inputName, 'mc_')] = [
+                    \PDO::PARAM_INT => (isset($ret[$inputName]) ? 1 : 0)
+                ];
+                break;
+            case 'host_stalOpts':
+                if (!empty($inputValue)) {
+                    $bindParams[':host_stalking_options'] = [
+                        \PDO::PARAM_STR => (($inputValue = filter_var(
+                            implode(",", array_keys($inputValue)),
+                            FILTER_SANITIZE_STRING
```
Comment (sc979) [inline www/include/configuration/configObject/host/DB-Func.php:2564]: SANITIZE_STRING is never false, but could be empty

**Review by sc979 — CHANGES_REQUESTED** [review]

File: www/include/configuration/configObject/host/DB-Func.php (hunk 2 above)
Comment (jeremyjaouen) [reply www/include/configuration/configObject/host/DB-Func.php:2513]: it could be false if you pass false (as a boolean)

**Review by callapa — COMMENTED** [review]
Why not also clean the fields in the table that already contain data?

**Comment by loiclau** [comment]
> Why not also clean the fields in the table that already contain data?

I don't know, we take the risk of emptying a mandatory field.

**Review by sc979 — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**